### PR TITLE
Cknieling/use cpp17

### PR DESCRIPTION
**Pull request by chr11115**
Add support for cpp17 and remove all boost

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -9,20 +9,18 @@ if(${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
...
 
 function(add_toolchain_definition TGT)
+	set_target_properties(${TGT} PROPERTIES CXX_STANDARD 17)
 	if(PLD_WINDOWS)
 		target_compile_definitions(${TGT} PRIVATE -DPLD_WINDOWS=1 -DPLD_TC_VC=1)
 	elseif(PLD_LINUX)
 		target_compile_definitions(${TGT} PRIVATE -DPLD_LINUX=1 -DPLD_TC_GCC=1)
-	elseif(PLD_MACOS)
-		target_compile_definitions(${TGT} PRIVATE -DPLD_MACOS=1 -DPLD_TC_CLANG=1)
+		target_compile_options(${PROJECT_NAME} PRIVATE -fvisibility=hidden -fvisibility-inlines-hidden)
```
Comment (mistafunk) [inline src/CMakeLists.txt:23]: FYI, should be `TGT` instead of `PROJECT_NAME`... already fixed in my dependent PR #180

**Review by mistafunk — CHANGES_REQUESTED** [review]
LGTM. Please first review #180 and then merge both if ok.

File: src/palladio/Utils.cpp [hunk 2]
```diff
@@ -58,14 +55,18 @@ void getCGBs(const ResolveMapSPtr& rm, std::vector<std::pair<std::wstring, std::
 	LOG_DBG << "   cgbList = '" << cgbList << "'";
 
 	std::vector<std::wstring> tok;
-	PLD_BOOST_NS::split(tok, cgbList, PLD_BOOST_NS::is_any_of(L";"), PLD_BOOST_NS::algorithm::token_compress_on);
-	for (const std::wstring& t : tok) {
-		if (t.empty())
+	std::wstringstream ss(cgbList);
```
[the remark this answers is not in the document]
Comment (chr11115) [reply src/palladio/Utils.cpp]: fixed

File: src/CMakeLists.txt (hunk 1 above)
Comment (chr11115) [reply src/CMakeLists.txt]: Thx, merged PR #180

**Pull request merged**